cmake/kokkos_arch.cmake

```diff
@@ -483,6 +471,39 @@ IF (KOKKOS_ARCH_POWER9)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_NATIVE)
+  INCLUDE(CheckCXXSymbolExists)
+
+  IF(NOT "${KOKKOS_COMPILE_OPTIONS}" STREQUAL "${KOKKOS_COMPILE_OPTIONS_SAVED}")
+    SET(CMAKE_REQUIRED_QUIET ON)
+    SET(CMAKE_REQUIRED_FLAGS "${KOKKOS_COMPILE_OPTIONS}")
+    SET(KOKKOS_COMPILE_OPTIONS_SAVED "${KOKKOS_COMPILE_OPTIONS}" CACHE INTERNAL "")
+
+    # Make sure to rerun the check if compile options have changed
+    UNSET(KOKKOS_COMPILER_HAS_AVX512)
```

Comment:
Please confirm that this does something.  I did not find any mention of the variable argument being defined influencing the way the check is done in the [doc](https://cmake.org/cmake/help/latest/module/CheckCXXSymbolExists.html).

Reply:
Yes, these are cached and the checks are not rerun otherwise.